code/game/atoms.dm

```diff
@@ -297,12 +297,12 @@ var/global/list/ghdel_profiling = list()
 		return 1
 	return
 
-/atom/proc/recursive_in_contents_of(var/atom/container, var/atom/searching_for = src)
-	if(isturf(searching_for))
-		return FALSE
+/atom/proc/recursive_in_contents_of(atom/container)
 	if(loc == container)
 		return TRUE
-	return recursive_in_contents_of(container, src.loc)
+	if(isarea(container))
+		return FALSE
+	return recursive_in_contents_of(container.loc)
 
```

Comment:
Check for a null container, otherwise runtime city

Reply:
done